https.go

```diff
@@ -117,8 +118,14 @@ func (proxy *ProxyHttpServer) handleHttps(w http.ResponseWriter, r *http.Request
 		if !hasPort.MatchString(host) {
 			host += ":80"
 		}
-
-		httpsProxy, err := httpsProxyAddr(r.URL, proxy.HttpsProxyAddr)
+		var httpsProxyURL string
+		if r.Header[PerRequestHTTPSProxyHeaderKey] != nil && r.Header[PerRequestHTTPSProxyHeaderKey][0] != "" {
```

Comment:
Could we change this to something like:

```
httpsProxyURL := proxy.HttpsProxyAddr
if r.Header.Get(PerRequestHTTPSProxyHeaderKey) != "" {
    httpsProxyURL = r.Header.Get(PerRequestHTTPSProxyHeaderKey) 
}

Reply:
We should also clear this header.

Reply:
Ack fixed! 

> We should also clear this header.
Not sure what this means?